storage/grpc_client.go

```diff
@@ -1052,7 +1060,8 @@ func (c *grpcStorageClient) NewRangeReader(ctx context.Context, params *newRange
 
 			// Receive the message into databuf as a wire-encoded message so we can
 			// use a custom decoder to avoid an extra copy at the protobuf layer.
-			err := stream.RecvMsg(&databuf)
+			databuf := &mem.BufferSlice{}
+			err := stream.RecvMsg(databuf)
```

Comment:
A mem.BufferSlice is a []Buffer. In general, it's not necessary to pass around pointers to slices. I'd change this to:

```
var databuf mem.BufferSlice
err := stream.RecvMsg(&databuf)
```

...and then also change readResponseDecoder.databufs to be a non-pointer BufferSlice. The field doesn't need to be pointer valued, and making it a plain BufferSlice will slightly simplify a bit of the readResponseDecoder code below.

(RecvMsg gets passed &databuf, because in this case we're using it as an out parameter.)